### ISSUE TITLE
When I track which thread is awakened by a certain thread, the CPU scheduling will not show which thread awakened.

### ISSUE THREAD

**Issue by 2292489630** [google/perfetto#158]
When I track which thread is awakened by a certain thread, the CPU scheduling will not show which thread awakened.
I see that in some examples, there will be a black origin on the black vertical line representing the next thread awakened by that thread. but I cant get it

**Comment by primiano** [comment]
I am sorry I cannot really understand the question. Can you reword it /  make some example / attach some screenshot / tell what you would expect to see instead?

**Comment by 2292489630** [comment]
this is a normal example
![image](https://user-images.githubusercontent.com/39506198/127626860-fa31aa22-d6cf-4739-9a9d-7621443c02b6.png)

but on my webpage,  when  I click on " running" , I cant find this thread was waked up by which thread .

![image](https://user-images.githubusercontent.com/39506198/127628259-bd0bbb71-6d5e-4615-b4bd-f860262c7cc7.png)

**Comment by primiano** [comment]
Scheduling latency events are emitted mainly for the "idle -> runnable" transition, because that's all about "which task did put another task in the condition of being schedulable again" (i.e. when you unblock a task sitting in a waiting chain).

the runnable -> running transition instead is not "unblocked by a thread" it's simply a scheduling decision: either the task previously running has suspended or it has run for too long and the scheduler decided to context switch to ensure fairness.

**Comment by 2292489630** [comment]
if I click on a running cut and turn to CPU , I can find this thread was waked by the other thread under normal circumstances 。
right？
but I cant find it like this ,it doesnt tell me these message.
![image](https://user-images.githubusercontent.com/39506198/127631180-2cc31c9c-5aff-4c11-a9d5-62e7e3673256.png)

**Comment by primiano** [comment]
>but I cant find it like this ,it doesnt tell me these message.

I cannot decode this sentence. I am not sure what you are looking for. If you scroll around you will find a black diamond that will tell you which CPU and at which time the release was caused.

Note that in some cases the cause of the wakeup will not map to any running task. That can happen when a task is unblocked in an interrupt context, in which case there is no task running on the scheduler.

**Comment by 2292489630** [comment]
yes , I just cant find this black diamond , do you know why?

**Comment by primiano** [comment]
No, that seems a bug.
Can you share the trace and the specific slice that is showing this behavior?

**Comment by LalitMaganti** [comment]
Closing as no further action can be taken without the trace - please feel free to reopen if the trace can be provided.
